In [15]:
# Importa la biblioteca pandas
import pandas as pd

# Carga los datos desde el archivo CSV
data = pd.read_csv('ds_salaries.csv')

# Calcula estadísticas básicas para la columna 'salary'
salary_stats = data['salary'].describe()

# Muestra las primeras filas del DataFrame para entender la estructura de los datos
data.head(9)

work_year experience_level employment_type                 job_title  \
0       2023               SE              FT  Principal Data Scientist   
1       2023               MI              CT               ML Engineer   
2       2023               MI              CT               ML Engineer   
3       2023               SE              FT            Data Scientist   
4       2023               SE              FT            Data Scientist   
5       2023               SE              FT         Applied Scientist   
6       2023               SE              FT         Applied Scientist   
7       2023               SE              FT            Data Scientist   
8       2023               SE              FT            Data Scientist   

   salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
0   80000             EUR          85847                 ES           100   
1   30000             USD          30000                 US           100   
2   25500             USD          25500                 US           100   
3  175000             USD         175000                 CA           100   
4  120000             USD         120000                 CA           100   
5  222200             USD         222200                 US             0   
6  136000             USD         136000                 US             0   
7  219000             USD         219000                 CA             0   
8  141000             USD         141000                 CA             0   

  company_location company_size  
0               ES            L  
1               US            S  
2               US            S  
3               CA            M  
4               CA            M  
5               US            L  
6               US            L  
7               CA            M  
8               CA            M

In [16]:
# Imprime las estadísticas descriptivas del salario
salary_stats


count    3.755000e+03
mean     1.906956e+05
std      6.716765e+05
min      6.000000e+03
25%      1.000000e+05
50%      1.380000e+05
75%      1.800000e+05
max      3.040000e+07
Name: salary, dtype: float64

In [28]:
# Promedio general
promedio = data['salary_in_usd'].mean()

# Desviación estándar
desviacion_estandar = data['salary_in_usd'].std()

# Quintiles
quintiles = data['salary_in_usd'].quantile([0.2, 0.4, 0.6, 0.8])

# Rango
rango = data['salary_in_usd'].max() - data['salary_in_usd'].min()

# Imprimir los resultados
print(" 1.- Promedio: {:.2f} USD\n".format(promedio))
print(" 2.- Desviación estándar: {:.2f} USD\n".format(desviacion_estandar))
print(" 3.- Quintiles:\n")
print(quintiles)
print("\n")
print(" 4.- Rango: {:.2f} USD".format(rango))


 1.- Promedio: 137570.39 USD

 2.- Desviación estándar: 63055.63 USD

 3.- Quintiles:

0.2     83972.8
0.4    120000.0
0.6    149850.0
0.8    185900.0
Name: salary_in_usd, dtype: float64


 4.- Rango: 444868.00 USD


In [5]:
# Calcula el salario promedio por nivel de experiencia
salarios_por_experiencia = data.groupby('experience_level')['salary_in_usd'].mean()

# Calcula el salario promedio por tipo de empleo
salarios_por_tipo_empleo = data.groupby('employment_type')['salary_in_usd'].mean()

# Calcula el salario promedio por grado de trabajo remoto
salarios_por_trabajo_remoto = data.groupby('remote_ratio')['salary_in_usd'].mean()

# Imprime los resultados
print("\nSalarios por Nivel de Experiencia:")
print(salarios_por_experiencia)
print("\nSalarios por Tipo de Empleo:")
print(salarios_por_tipo_empleo)
print("\nSalarios por Trabajo Remoto:")
print(salarios_por_trabajo_remoto)



Salarios por Nivel de Experiencia:
experience_level
EN     78546.284375
EX    194930.929825
MI    104525.939130
SE    153051.071542
Name: salary_in_usd, dtype: float64

Salarios por Tipo de Empleo:
employment_type
CT    113446.900000
FL     51807.800000
FT    138314.199570
PT     39533.705882
Name: salary_in_usd, dtype: float64

Salarios por Trabajo Remoto:
remote_ratio
0      144316.202288
50      78400.687831
100    136481.452830
Name: salary_in_usd, dtype: float64


1. Interpretación de los Indicadores Generales:
Promedio: El salario promedio en el conjunto de datos es de aproximadamente 137,570.39 USD.
Desviación Estándar: La desviación estándar es de aproximadamente 63,055.63 USD, lo que indica una variabilidad significativa en los salarios.
Quintiles: La distribución de los salarios muestra que el 20% de los empleados ganan menos de 83,972.8 USD, el 40% ganan menos de 120,000 USD, el 60% ganan menos de 149,850 USD y el 80% ganan menos de 185,900 USD.
Rango: La diferencia entre el salario más bajo y el más alto es de 444,868.00 USD.

2. Interpretación de los Salarios Agrupados:
Por Nivel de Experiencia:

Los empleados con nivel ejecutivo (EX) tienen el salario promedio más alto, alrededor de 194,930.93 USD, seguidos por empleados con nivel senior (SE) con un promedio de aproximadamente 153,051.07 USD. Los niveles de entrada (EN) tienen el salario promedio más bajo, aproximadamente 78,546.28 USD.
Por Tipo de Empleo:

Empleados a tiempo completo (FT) tienen el salario promedio más alto con aproximadamente 138,314.20 USD, mientras que empleados a tiempo parcial (PT) tienen el salario promedio más bajo, alrededor de 39,533.71 USD.
Por Grado de Trabajo Remoto:

Empleados que trabajan completamente de forma remota (remote_ratio 100) tienen el salario promedio más alto, aproximadamente 136,481.45 USD. Aquellos con un trabajo remoto parcial (remote_ratio 50) tienen un salario promedio más bajo, alrededor de 78,400.69 USD.
Interpretación:
Variabilidad: La alta desviación estándar y el rango amplio indican una variabilidad significativa en los salarios.

Tendencia Central: Aunque los empleados con nivel ejecutivo (EX) tienen el salario promedio más alto, los salarios para niveles de entrada (EN) varían considerablemente. Esto se refleja en la alta desviación estándar y en la diferencia entre los quintiles 20% y 80%. En términos de tipo de empleo, los empleados a tiempo completo (FT) tienen una variabilidad relativamente baja en comparación con otros tipos de empleo.

Influencia del Trabajo Remoto: Aunque los empleados completamente remotos tienen un salario promedio alto, hay variabilidad significativa, como se ve en la desviación estándar. Esto podría deberse a factores como la industria y la experiencia individual.

